library/src/scala/runtime/quoted/Unpickler.scala

```diff
@@ -1,24 +1,27 @@
 package scala.runtime.quoted
 
-import scala.internal.quoted.{TastyExpr, TastyType}
 import scala.quoted.{Expr, QuoteContext, Type}
+import scala.runtime.quoted.Unpickler.{PickledExpr, PickledType}
 
 /** Provides methods to unpickle `Expr` and `Type` trees. */
 object Unpickler {
 
-  /** Representation of pickled trees. For now a List[String],
-   *  but it should be changed to some kind of TASTY bundle.
-   */
-  type Pickled = List[String]
+  type PickledExpr = List[String]
+  type PickledExprArgs = Seq[Seq[Any] => ((given QuoteContext => Expr[Any]) | Type[_])]
+
+  type PickledType = List[String]
+  type PickledTypeArgs = Seq[Seq[Any] => Type[_]]
 
   /** Unpickle `repr` which represents a pickled `Expr` tree,
    *  replacing splice nodes with `args`
    */
-  def unpickleExpr[T](repr: Pickled, args: Seq[Seq[Any] => ((given QuoteContext => Expr[Any]) | Type[_])]): given QuoteContext => Expr[T] = new TastyExpr[T](repr, args)
+  def unpickleExpr[T](repr: PickledExpr, args: PickledExprArgs): given QuoteContext => Expr[T] =
+    the[QuoteContext].unpickleExpr(repr, args)
```

Comment:
If I were correct, the code will be expanded as follows:

```Scala
qctx => qctx.unpickleExpr(repr, args).apply(qctx)
```

It seems we can change the return type of `QuoteContext.unpickleExpr` to be just `Expr[T]`.

Reply:
Yes, `QuoteContext.unpickleExpr` does not need the quote context argument.